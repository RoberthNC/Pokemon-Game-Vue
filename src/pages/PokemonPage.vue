<template>
  <h1>¿Quién es este Pokémon?</h1>
  <PokemonImage :pokemonId="4" :showPokemon="true" />
  <PokemonOptions :pokemons="pokemonArr" />
</template>

<script>
import PokemonImage from "@/components/PokemonImage.vue";
import PokemonOptions from "@/components/PokemonOptions.vue";
import getPokemonOptions from "@/helpers/getPokemonOptions.js";

export default {
  name: "PokemonPage",
  components: {
    PokemonImage,
    PokemonOptions,
  },
  data() {
    return {
      pokemonArr: [],
    };
  },
  methods: {
    async mixPokemonArray() {
      this.pokemonArr = await getPokemonOptions();
      console.log(this.pokemonArr);
    },
  },
  mounted() {
    this.mixPokemonArray();
  },
};
</script>
