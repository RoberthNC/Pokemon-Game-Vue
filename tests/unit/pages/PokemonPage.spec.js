import { pokemons } from "../mocks/pokemons.mock";
import { shallowMount, mount } from "@vue/test-utils";
import PokemonPage from "@/pages/PokemonPage.vue";

describe("PokemonPage component", () => {
  let wrapper;

  beforeEach(() => {
    wrapper = shallowMount(PokemonPage);
  });

  test("Should match with snapshot", () => {
    // expect(wrapper.html()).toMatchSnapshot();
  });

  test("Should call mixPokemonArray when the componentn is mounted", () => {
    const mixPokemonArray = jest.spyOn(PokemonPage.methods, "mixPokemonArray");
    shallowMount(PokemonPage);
    expect(mixPokemonArray).toHaveBeenCalled();
  });

  test("Should match with snapshot when pokemons load", () => {
    const wrapper = shallowMount(PokemonPage, {
      data() {
        return {
          pokemonArr: pokemons,
          pokemon: pokemons[0],
          showPokemon: false,
          showAnswer: false,
          message: "",
        };
      },
    });
    expect(wrapper.html()).toMatchSnapshot();
  });

  test("Should show PokemonImage and PokemonOptions components", () => {
    const wrapper = shallowMount(PokemonPage, {
      data() {
        return {
          pokemonArr: pokemons,
          pokemon: pokemons[0],
          showPokemon: false,
          showAnswer: false,
          message: "",
        };
      },
    });
    expect(wrapper.find("pokemon-image-stub").exists()).toBeTruthy();
    expect(wrapper.find("pokemon-options-stub").exists()).toBeTruthy();
    expect(wrapper.find("pokemon-image-stub").attributes("pokemonid")).toBe(
      "1"
    );
    expect(
      wrapper.find("pokemon-options-stub").attributes("pokemons")
    ).toBeTruthy();
  });

  test("Testing on checkAnswer", async () => {
    const wrapper = shallowMount(PokemonPage, {
      data() {
        return {
          pokemonArr: pokemons,
          pokemon: pokemons[0],
          showPokemon: false,
          showAnswer: false,
          message: "",
        };
      },
    });
    await wrapper.vm.checkAnswer(1);
    expect(wrapper.find("h2").exists()).toBeTruthy();
    expect(wrapper.vm.showPokemon).toBeTruthy();
    expect(wrapper.find("h2").text()).toBe(
      `Correcto! Es un ${pokemons[0].name}`
    );
    await wrapper.vm.checkAnswer(10);
    expect(wrapper.vm.message).toBe(`Oops! Era un ${pokemons[0].name}`);
  });
});
