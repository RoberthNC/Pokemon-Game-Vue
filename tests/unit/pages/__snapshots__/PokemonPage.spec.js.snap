// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PokemonPage component Should match with snapshot when pokemons load 1`] = `
<div>
  <h1>¿Quién es este Pokémon?</h1>
  <pokemon-image-stub pokemonid="1" showpokemon="false"></pokemon-image-stub>
  <pokemon-options-stub pokemons="[object Object],[object Object],[object Object],[object Object]"></pokemon-options-stub>
  <!--v-if-->
</div>
`;
